lib/interface.ts

```diff
@@ -14,3 +14,14 @@ export interface IOurResponse extends restify.Response {
     sendEnd?: (status: any, message: string) => Promise<void>;
     sendError?: (err: any, where: string, reason?: any) => Promise<void>;
 }
+
+export interface ISocket {
+    log: bunyan.Logger; // Add bunyan logger to socket
+    blpSession: BAPI.Session;  // Add blpSession to socket
+    isConnected(): boolean;
+    getIP(): string;
+    getCert(): any;
+    emit(name: string, ...args: any[]): void;
+    disconnect(): void;
+    on(event: string, listener: Function): void;
```

Comment:
I think this interface can just extend `EventEmitter` from the node `events` module.